controllers/topology/object_builders_test.go

```diff
@@ -430,3 +434,110 @@ func (f *fakeControlPlane) Obj() *unstructured.Unstructured {
 	}
 	return obj
 }
+
+type fakeCluster struct {
+	namespace       string
+	name            string
+	class           clusterv1.ClusterClass
+	controlPlaneRef *corev1.ObjectReference
+	infraRef        *corev1.ObjectReference
+}
+
+func newFakeCluster(namespace, name string) *fakeCluster {
+	return &fakeCluster{
+		namespace: namespace,
+		name:      name,
+	}
+}
+func (f *fakeCluster) withClass(class clusterv1.ClusterClass) *fakeCluster {
+	f.class = class
+	return f
+}
+
+func (f *fakeCluster) withControlPlaneRef(reference *corev1.ObjectReference) *fakeCluster {
```

Comment:
For consistency with the existing builders we should make this method public and pass in an unstructured object, not a reference
```suggestion
func (f *fakeCluster) WithControlPlane(reference *corev1.ObjectReference) *fakeCluster {
```

See e.g. https://github.com/kubernetes-sigs/cluster-api/blob/0887a1d8b6e99dfca29ddede4d449c0bf7a71335/controllers/topology/object_builders_test.go#L109-L154 for a similar change in #5052.

Same applies to other changes to the object builders in this PR

Reply:
I think the latest version has resolved this.

Reply:
Thanks!